src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TplEventSource.cs

```diff
@@ -538,7 +538,7 @@ public void NewID(int TaskID)
         public void IncompleteAsyncMethod(IAsyncStateMachineBox stateMachineBox)
         {
             System.Diagnostics.Debug.Assert(stateMachineBox != null);
-            if (IsEnabled(EventLevel.Warning, Keywords.AsyncMethod))
+            if (IsEnabled() && IsEnabled(EventLevel.Warning, Keywords.AsyncMethod))
```

Comment:
Why are these two changes needed?

Reply:
This one is needed because the result of `IsEnabled(EventLevel, Keywords)` can't be inlined (see my other [comment](https://github.com/dotnet/runtime/pull/38129#discussion_r443060889)). And this method is called from a Finalizer.

For `TaskCompleted`, it is needed because of https://github.com/mono/linker/issues/1284. The calls to this method aren't being trimmed by the linker even though logically they are behind `IsEnabled()` checks through local variables.

I figured this would be acceptable since the other methods in this class follow this same pattern and call both.